pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/OfflineSegmentAssignment.java

```diff
@@ -53,62 +49,57 @@ protected int getReplication(TableConfig tableConfig) {
   public List<String> assignSegment(String segmentName, Map<String, Map<String, String>> currentAssignment,
       Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap) {
     InstancePartitions instancePartitions = instancePartitionsMap.get(InstancePartitionsType.OFFLINE);
+    // Gets Segment assignment strategy for instance partitions
+    SegmentAssignmentStrategy segmentAssignmentStrategy = SegmentAssignmentStrategyFactory
+        .getSegmentAssignmentStrategy(_helixManager, _tableConfig, instancePartitions, InstancePartitionsType.OFFLINE);
     Preconditions.checkState(instancePartitions != null, "Failed to find OFFLINE instance partitions for table: %s",
```

Comment:
Let's move this line up so that we call the precondition and then pass `instancePartitions` to the factory. Then, we will have the guarantee that `instanceParittions` is always not null.

Reply:
done